ui/apps/platform/src/hooks/useFetchReport.ts

```diff
@@ -1,31 +1,104 @@
 import { useEffect, useState } from 'react';
+import { fetchAccessScopes } from 'services/AccessScopesService';
+import { getCollection } from 'services/CollectionsService';
 
 import { fetchReportById } from 'services/ReportsService';
 import { ReportConfiguration } from 'types/report.proto';
+import useFeatureFlags from './useFeatureFlags';
 
-type Result = { isLoading: boolean; report: ReportConfiguration | null; error: string | null };
+export type ReportScope = {
+    // The 'AccessControlScope' type is deprecated and should be able to be removed in the release after 3.74
+    type: 'CollectionScope' | 'AccessControlScope';
+    id: string;
+    name: string;
+    description: string;
+};
 
-const defaultResultState = { report: null, error: null, isLoading: true };
+type Result = {
+    isLoading: boolean;
+    report: ReportConfiguration | null;
+    reportScope: ReportScope | null;
```

Comment:
`reportScope` was added to the `Result` type here, instead of making it a child property of `Result['report']`, since the scope of the `ReportConfiguration` type is much larger and this limits the scope of the change.

Reply:
Delightful pun to reward the reviewer:

> scope of the change